### ISSUE TITLE
Filter search results by service type / taxonomy

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/ServiceNet#531]
As always, I'm open to suggestions on how we might most efficiently and elegantly achieve this.

The goal for this task is to introduce an additional filtering option similar to the geo filtering we already have. This filter will allow the user to segment organizations that have services in a specific 'service type' or 'service taxonomy'. For example - show me housing related resources in Oakland, etc..

Unsurprisingly partners have different ways of recording these:

* UWBA and Eden use the resourceDetail{taxonomy} array which joins to the taxonomyCodeInfo object which can be grabbed from the /Resource/Taxonomy endpoint.

* LAAC only collects Legal resources - so those resources will always be of type 'Legal'. They also have a 'Legal Problem' column that gets a little more specific category.

* SMC has a taxonomyID in services.csv - the translation to text is available here: https://api.smc-connect.org/categories

* ShelterTech - under the {categories} json element they have both Top level and non top level categories

* HealthLeads - services_taxonomy.csv links to taxonomy.csv

So the most challenging component is finding commonality in these various taxonomies. 

I think the ideal endstate would be for each user to see the filtering options in their own taxonomical terms, and on the backend we map their terms to the corresponding terms in partner systems. In this way they dont get boggeds down trying to keep track of all the various ways people categorize things

Some part of that mapping will no doubt be manual - but i think we can look at our matches and highly correlated taxonomical terms between systems to give us a head start.

**Comment by johnhbenetech** [comment]
The first component of this task is to make sure the various taxonomies and their corresponding terms (like their text description not just ID) are available to the application. This may require an additional table, but ill leave that up to you.

The second component is just to add a multiselect filter (similar to the county/city filters) that pulls the logged in accounts corresponding taxonomy terms for choices.

**Comment by NeilMBenetech** [comment]
I suspect the Taxonomy "rosetta stone is a prerequisite for this task. https://github.com/benetech/ServiceNet/issues/319

**Comment by johnhbenetech** [comment]
We can put it on the main view without #319 - since we are only filtering their own records using their own vocabulary.

Once discovery is in place (without taxonomy filtering) we would do that translation work and then add it in.

We can certainly get a start on our side of #319 (the nasty manual parts :D )

**Comment by NeilMBenetech** [comment]
I've incorporated some of this thinking in the Vision doc I just shared with you. You'll see I'm suggesting that Discovery really is just a component of Search and that taxonomy filtering is something I think that should exist no matter what kind of Search one is doing. Let's debate that thinking in that "bigger picture" document first and then come back to this item to figure out the ramifications before we put it into development.

**Comment by bognasoldev89** [comment]
Confirmed that filtering by taxonomy works for admin and other users.

**Comment by NeilMBenetech** [comment]
@johnhbenetech This appears to be working but I can't fully confirm because the "Service Taxonomy" field in Service records is blank in the numerous ones I looked at. 

Also, similar to my feedback today on https://github.com/benetech/ServiceNet/issues/340 , the search results should list the Services that matched for the taxonomy filter, not the parent Organization. And when you click on a Service in search results, the one(s) matching the search criteria should be pre-selected so you don't have to laboriously cycle through them.

**Comment by NeilMBenetech** [comment]
@johnhbenetech since the search appears to be working, we can consider this item closed. I'll create a new item to address the "Service Taxonomy" being blank when viewing details of a record.

I have also created a new task for displaying search results at the Service level.
